tests/TypeTests.hs

```diff
@@ -11,8 +11,15 @@ import GraphQL.Schema
 
 -- Examples taken from the spec
 
+data DogCommandEnum = Sit | Down | Heel
+
+instance GraphQLEnum DogCommandEnum where
+  enumValues = ["SIT", "DOWN", "HEEL"]
+  enumToValue _ = undefined
+  enumFromValue _ = undefined
+
 -- Alternative might be a sum type with deriving Generic and 0-arity constructors?
-type DogCommand = Enum "DogCommand" '["SIT", "DOWN", "HEEL"]
+type DogCommand = Enum "DogCommand" DogCommandEnum
```

Comment:
The ergonomics of the API definition are worse, but when you have to use the Enum in your own code I think it's better overall. I've been on the fence here for a while but after the Git and Filesystem examples I'm pretty sure that we want the type-class enum (+ DeriveGeneric)